Yosemite/YosemiteTests/Tools/Payments/WooPaymentsDepositServiceTests.swift

```diff
@@ -0,0 +1,114 @@
+import XCTest
+@testable import Yosemite
+@testable import Networking
+
+final class WooPaymentsDepositServiceTests: XCTestCase {
+    var service: WooPaymentsDepositService!
+    var mockNetwork: MockNetwork!
+
+    override func setUp() {
+        super.setUp()
+        mockNetwork = MockNetwork()
+        service = WooPaymentsDepositService(siteID: 12345, network: mockNetwork)
+    }
+
+    override func tearDown() {
+        mockNetwork = nil
+        service = nil
+        super.tearDown()
+    }
+
+    func test_fetchDepositsOverview_returns_one_model_per_response_element() async {
+        // Given
+        mockNetwork.simulateResponse(requestUrlSuffix: "payments/deposits/overview-all", filename: "deposits-overview-all")
+
+        do {
+            // When
+            let depositsOverviews = try await service.fetchDepositsOverview()
+
+            // Then
+            assertEqual(2, depositsOverviews.count)
+        } catch {
```

Comment:
For this and the other caught errors in this file, do we actually need to catch them in the test function, or can we just mark these functions as `throws`? The system will handle any errors that will be generated by the above code, and will automatically turn any such errors into proper test failures.
But if you want to explicitly catch them and log an error, then it's fine as it is 👍

Reply:
Yeah, I think that catching makes it slightly more readable when they fail, and allows the test runner to keep going. I wouldn't neccesarily bother if I were writing it all out by hand, but they're pretty simple tests and it's a nice addition that ChatGPT put in.